File: mario_java/src/Background.java
//**********************************************************************************************************************
// Background
//**********************************************************************************************************************
import java.awt.*;

public class Background extends Sprite
{
    /*
    x offset: somewhere behind model's x = 0, so you can move left a bit
    y offset: 0
    background sprite width: (width of .png)
    background sprite height: (height of .png)
    */
    Background() { super(-250, 0, 2500, 650); }
    Background(Json j) { super(j); }

    @Override
    public boolean isBackground()
    {
        return true;
    }

    @Override
    void update() { }

    @Override
    void draw(Graphics g)
    {
        if(isBackground())
        {
            g.drawImage(images[0], this.x - Sprite.scrollPos / 2, this.y, null);
        }
    }
}

File: mario_client_server/background.js
// Screen width and height (of the game, NOT the entire web page)
let screenWidth = 2500;
let screenHeight = 500;

let screenStartPos = -200;

//=====================================================================================================================
// Background "class"
//=====================================================================================================================
class Background extends Sprite
{
    constructor(inputX, inputY)
    {
        super(inputX, inputY, screenWidth, screenHeight);

        this.image1 = new Image();
        this.image1.src = "background.png";

        this.type = "background";
    }

    //=================================================================================================================
    // Update background mechanics
    //=================================================================================================================
    update()
    {
        this.scrollPos = masterScrollPos;
    }

    //=================================================================================================================
    // Draw the background
    //=================================================================================================================
    draw(context)
    {
        context.drawImage(this.image1, this.xPos - masterScrollPos, this.yPos, this.width, this.height);
    }
}